service/index.ts

```diff
@@ -10,325 +10,323 @@ import { DaoFactory } from './factory/DaoFactory';
-        const user = await getUserFromRequest(req)
-        const roomId = req.params.id
-        const room = await roomDAO.getRoomById(roomId);
-
-        if (!room) {
-            res.status(404).send({ msg: `Room ${roomId} does not exist` })
-            return
-        }
-        const isOwner = room.owner === user!.username
-
-        if (!isOwner) {
-            res.status(403).send({ msg: 'User is not owner of room' })
-            return
-        }
-
-        if (room.state !== 'open') {
-            res.status(409).send({ msg: 'Room is not open' })
-            return
-        }
-
-        await roomDAO.closeRoom(roomId);
-
-        const {sortedOptions, sortedTotals} = calculateVoteResult(room.votes)
-        const result = await historyDAO.createResult(user!.username, sortedOptions, sortedTotals);
-
-        res.status(200).send({ resultsId: result._id })
-    })
-
-    secureApiRouter.get('/results/:id', async (req: Request, res: Response) => {
-        const resultsId = req.params.id
-        const result = await historyDAO.getResult(resultsId);
-
-        if (!result) {
-            res.status(404).send({ msg: `Result does not exist` })
-            return
-        }
-
-        res.status(200).send({ results: result.sortedOptions, totals: result.sortedTotals })
-    })
-
-    secureApiRouter.get('/history', async (req: Request, res: Response) => {
-        const user = await getUserFromRequest(req)
-        const history = await historyDAO.getHistory(user!.username);
-
-        res.status(200).send({ history })
-    })
+  secureApiRouter.use(async (req: Request, res: Response, next: NextFunction) => {
+    const user = await getUserFromRequest(req);
+    if (user) {
+      next();
+    } else {
+      res.status(401).send({ msg: 'Unauthorized' });
+    }
+  });
 
-    app.use(function(err: Error, _req: Request, res: Response, _next: NextFunction) {
-        res.status(500).send({ type: err.name, message: err.message });
-    });
+  secureApiRouter.post('/room', async (req: Request, res: Response) => {
+    const user = await getUserFromRequest(req)
+    const newRoom = await roomDAO.createRoom(user!.username);
 
-    app.use((_req: Request, res: Response) => {
-        res.sendFile('index.html', { root: 'public' });
-    });
+    res.status(201).send({ id: newRoom._id, code: newRoom.code })
+  })
+
+  secureApiRouter.get('/room/:id', async (req: Request, res: Response) => {
+    const user = await getUserFromRequest(req)
+    const roomId = req.params.id
+    const room = await roomDAO.getRoomById(roomId);
+
+    if (!room) {
+      res.status(404).send({ msg: `Room ${roomId} does not exist` })
+      return
+    }
+
+    if (room.state !== 'open') {
+      res.status(409).send({ msg: 'Room is not open' })
+      return
+    }
+
+    await roomDAO.addParticipantToRoom(room.code, user!.username);res.status(200).send({ ...room, isOwner: room.owner === user!.username })
```

Comment:
nit: these should probably be on separate lines
```suggestion
    await roomDAO.addParticipantToRoom(room.code, user!.username);
    res.status(200).send({ ...room, isOwner: room.owner === user!.username })
```